packages/salesforcedx-vscode-core/src/context/workspaceContext.ts

```diff
@@ -54,3 +63,136 @@ export class WorkspaceContext {
+    if (!configAggregator) {
+      configAggregator = await this.createConfigAggregator();
+      this.configAggregators.set(getRootWorkspacePath(), configAggregator);
+    }
+    return configAggregator;
+  }
+
+  public async getSfdxConfigAggregator(): Promise<ConfigAggregator> {
+    let sfdxConfigAggregator = this.sfdxConfigAggregators.get(
+      getRootWorkspacePath()
+    );
+    if (!sfdxConfigAggregator) {
+      sfdxConfigAggregator = await this.createConfigAggregator({ sfdx: true });
+      this.sfdxConfigAggregators.set(
+        getRootWorkspacePath(),
+        sfdxConfigAggregator
+      );
+    }
+    return sfdxConfigAggregator;
+  }
+
+  public async getGlobalConfigAggregator(): Promise<ConfigAggregator> {
+    if (!this.globalConfigAggregator) {
+      this.globalConfigAggregator = await this.createConfigAggregator({
+        globalValuesOnly: true
+      });
+    }
+    return this.globalConfigAggregator;
+  }
+
+  public async reloadConfigAggregators() {
+    console.log(
+      'The .sfdx config file has changed.  Reloading ConfigAggregator values in the salesforcedx-vscode-core package.'
+    );
+    // Force ConfigAggregator to load the most recent values from
+    // the config file.  This prevents an issue where ConfigAggregator
+    // can returned cached data instead of the most recent data.
+    const configAggregator = this.configAggregators.get(getRootWorkspacePath());
+    if (configAggregator) await configAggregator.reload();
+
+    const sfdx = this.sfdxConfigAggregators.get(getRootWorkspacePath());
+    if (sfdx) await sfdx.reload();
+  }
+
+  private async createConfigAggregator(
+    options: ConfigAggregatorOptions = {
+      sfdx: false,
+      globalValuesOnly: false
+    }
+  ): Promise<ConfigAggregator> {
+    let configAggregator;
+    const currentWorkingDirectory = process.cwd();
+    if (options.globalValuesOnly) {
+      ConfigAggregatorProvider.ensureProcessIsRunningUnderUserHomeDir(
+        currentWorkingDirectory
+      );
+    } else {
+      // Change the current working directory to the project path,
+      // so that ConfigAggregator reads the local project values
+      ConfigAggregatorProvider.ensureProcessIsRunningUnderProjectRoot(
+        currentWorkingDirectory
+      );
+    }
+    try {
+      configAggregator = options.sfdx
+        ? await SfdxConfigAggregator.create()
+        : await ConfigAggregator.create();
+    } finally {
+      // Change the current working directory back to what it was
+      // before returning.
+      // Wrapping this in a finally block ensures that the working
+      // directory is switched back to what it was before this method
+      // was called if SfdxConfigAggregator.create() throws an exception.
+      process.chdir(currentWorkingDirectory);
+    }
+    return configAggregator;
+  }
+
+  private static ensureProcessIsRunningUnderUserHomeDir(path: string) {
+    const userHomePath = '/';
```

Comment:
Just calling out this says user path, but '/' isn't a home path.  It's like a base file system path.  If this is an existing flow (or what sfdx does) then fine, but some users may not have permission to navigate to that directory

Reply:
This is the default dir that VSCE runs processes under.  If the caller wants a global values only configAgg, then the process needs to be running under the Users root dir (same dir where the global .sf and .sfdx dirs are) when the ConfigAgg is created.

Reply:
![Screen Shot 2022-08-25 at 6 04 24 PM](https://user-images.githubusercontent.com/76090802/186783725-9b18e336-c1d0-40eb-b0a7-cceb9287a27b.png)

So does ConfigAgg do the global lookup in the user home dir if no config directories are found?  Or does it need to be in the users home dir in order to find and load the globals?

In the screenshot I navigated to '/' first and there are no sfdx config dirs.  Then the second dir is my home dir where my global config dirs exits.  

I suspect if no config directories are found then configAgg will check the user home space for global config dirs so we're good to go. 

One naming nit here to avoid confusing might be to rename the method to `ensureCurrentDirectoryOutsideProject` or something...stupid naming.